ReSharper.FSharp/src/FSharp/FSharp.Psi.Daemon/src/Stages/PipeChainTypeHintStage.fs

```diff
@@ -54,19 +90,46 @@ type PipeOperatorVisitor(sameLinePipeHints: SameLinePipeHints) =
 type PipeChainHighlightingProcess(logger: ILogger, fsFile, settings: IContextBoundSettingsStore, daemonProcess: IDaemonProcess) =
     inherit FSharpDaemonStageProcessBase(fsFile, daemonProcess)
 
-    let adornExprs logKey (exprs : (IReferenceExpr * ITreeNode)[]) =
+    let isRightPipeArgApplicable (expr: ITreeNode) (pipeResultType: FSharpType) =
+        if not (isUnit pipeResultType) then true else
+
+        let rec skipFunArgs (t: FSharpType) =
+            if t.IsFunctionType then skipFunArgs t.GenericArguments[1] else t
```

Comment:
I'm not sure I can easily follow the logic here. Could you add some comments/examples, please?